packages/common/src/extensions/slickCheckboxSelectColumn.ts

```diff
@@ -461,6 +465,18 @@ export class SlickCheckboxSelectColumn<T = any> {
     }
   }
 
+  protected handleRowsChanged() {
+    let disableSelectAll = false;
+
+    if (!this._addonOptions.applySelectOnAllPages && typeof this._selectableOverride === 'function') {
+      const filteredItems = this._dataView.getFilteredItems();
+      // disable select all checkbox if 1) nothing to select or 2) no item can be selected
+      // requires applySelectOnAllPages to be false, as we don't want to make assumptions on data on other pages
+      disableSelectAll = filteredItems.length === 0 || filteredItems.every(dataItem => !this.checkSelectableOverride(0, dataItem, this._grid));
```

Comment:
I'm not too crazy about this approach, if I understand correctly it loop through the entire dataset every single time a row changes, which is in some way really bad performance wise. Imagine a grid with 100K rows this would keep the CPU a little too busy... There must be a way to do it in 1 pass instead of multiple passes every time a row changes. 

Another thing to note also would be that if we have 1 valid row, then there's no reason to investigate further

Reply:
What about using a State object to keep the count of enabled/disabled like `state = { enabled: 333, disabled: 1 }` and if you want to restart from scratch then also add a reset function like `forceStateCheck()`. If that doesn't then another solution could be to use a `Set` or a `Map` to keep just the `id`/`state` pair, at least that is a lot more performant than looping through the entire dataset, updating the Map or Set is very quick, it also gets rid of duplicate since by default it doesn't allow duplicates.